docs/source/examples/Widget Asynchronous.ipynb

```diff
@@ -282,6 +282,213 @@
+       "model_id": "c5a24b1c6cbe4089bbdc5d36c6edac4f",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/html": [
+       "<p>Failed to display Jupyter Widget of type <code>interactive</code>.</p>\n",
+       "<p>\n",
+       "  If you're reading this message in Jupyter Notebook or JupyterLab, it may mean\n",
+       "  that the widgets JavaScript is still loading. If this message persists, it\n",
+       "  likely means that the widgets JavaScript library is either not installed or\n",
+       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
+       "  Widgets Documentation</a> for setup instructions.\n",
+       "</p>\n",
+       "<p>\n",
+       "  If you're reading this message in another notebook frontend (for example, a static\n",
+       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
+       "  it may mean that your frontend doesn't currently support widgets.\n",
+       "</p>\n"
+      ],
+      "text/plain": [
+       "interactive(children=(Text(value='WHATEVER', description='Whatever'), Output()), _dom_classes=('widget-interact',))"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import datetime\n",
+    "import threading\n",
+    "import time\n",
+    "\n",
+    "from IPython.display import (\n",
+    "    HTML,\n",
+    "    display,\n",
+    ")\n",
+    "import ipywidgets\n",
+    "\n",
+    "def alpha(whatever):\n",
+    "\n",
+    "    def beta(something, container, timerlength=1.5, interval=0.1):\n",
+    "        ctr = 0\n",
+    "        \n",
+    "        pre_output = ipywidgets.Output()\n",
+    "        with pre_output:\n",
+    "            display('Start beta() at {}'.format(time.time()))\n",
+    "        container.children += (pre_output, )\n",
+    "            \n",
+    "        while ctr < timerlength:\n",
+    "            time.sleep(interval)\n",
+    "            ctr += interval\n",
+    "            \n",
+    "        post_output = ipywidgets.Output()\n",
+    "        with post_output:\n",
+    "            display(\"beta() One {} at {}\".format(something, time.time()))\n",
+    "            display(\"beta() Two {} at {}\".format(something, time.time()))\n",
+    "            display(\"beta() Thr {} at {}\".format(something, time.time()))\n",
+    "            display(\"beta() Fou {} at {}\".format(something, time.time()))\n",
+    "            display(\"beta() Fiv {} at {}\".format(something, time.time()))        \n",
+    "        button2 = ipywidgets.Button(description=\"Button #2\")\n",
+    "        container.children += (post_output, button2)\n",
+    "    \n",
+    "    container = ipywidgets.VBox()\n",
+    "    button = ipywidgets.Button(description=\"Button #1\")\n",
+    "    container.children += (button, )\n",
+    "\n",
+    "    display('I am in alpha() at {}'.format(time.time()))\n",
```

Comment:
Explicitly saying 'In main thread', rather than 'I am in alpha', seems like it would help illustrate that you can interact with the widget both from the main thread and the child.